app/scripts/controllers/swaps.js

```diff
@@ -104,23 +104,34 @@ const initialState = {
 };
 
 export default class SwapsController {
-  constructor({
-    getBufferedGasLimit,
-    networkController,
-    provider,
-    getProviderConfig,
-    getTokenRatesState,
-    fetchTradesInfo = defaultFetchTradesInfo,
-    getCurrentChainId,
-    getEIP1559GasFeeEstimates,
-    onNetworkStateChange,
-  }) {
+  constructor(
+    {
+      getBufferedGasLimit,
+      networkController,
+      provider,
+      getProviderConfig,
+      getTokenRatesState,
+      fetchTradesInfo = defaultFetchTradesInfo,
+      getCurrentChainId,
+      getEIP1559GasFeeEstimates,
+      onNetworkStateChange,
+    },
+    state,
+  ) {
     this.store = new ObservableStore({
-      swapsState: { ...initialState.swapsState },
+      swapsState: {
+        ...initialState.swapsState,
+        swapsFeatureFlags: state?.swapsState?.swapsFeatureFlags || {},
```

Comment:
Why was the `swapsFeatureFlags` property explicitly added here? This would always initialize the feature flag state to what it was during the last user session. Is that intended?

Reply:
Glad you asked! The reason for this is the following: if we would turn off the feature flag for this new UI redesign (in case there is a blocking issue), we would anyway [show the new redesign by default](https://github.com/MetaMask/metamask-extension/pull/19169/files#diff-2493772b1f68622e8d2eae0660b50d2765884331c60418c86e5315389fb2fd8eR345) before redirecting to the previous design once we get a response from the `/featureFlags` endpoint, which is not user-friendly. This way we will remember which feature flags were used last time and use those by default, before we get fresh ones from the `/featureFlags` endpoint.